dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/export/relationship/HibernateRelationshipStore.java

```diff
@@ -87,97 +88,121 @@ public HibernateRelationshipStore(
-        getRelationshipTypedQuery(trackedEntity, queryParams, null);
 
-    return getList(relationshipTypedQuery);
+    return relationshipsList(trackedEntity, queryParams, null);
   }
 
   @Override
   public List<Relationship> getByEnrollment(
       Enrollment enrollment, RelationshipQueryParams queryParams) {
-    TypedQuery<Relationship> relationshipTypedQuery =
-        getRelationshipTypedQuery(enrollment, queryParams, null);
-
-    return getList(relationshipTypedQuery);
+    return relationshipsList(enrollment, queryParams, null);
   }
 
   @Override
   public List<Relationship> getByEvent(Event event, RelationshipQueryParams queryParams) {
-    TypedQuery<Relationship> relationshipTypedQuery =
-        getRelationshipTypedQuery(event, queryParams, null);
-
-    return getList(relationshipTypedQuery);
+    return relationshipsList(event, queryParams, null);
   }
 
   @Override
   public Page<Relationship> getByTrackedEntity(
       TrackedEntity trackedEntity,
-      RelationshipQueryParams queryParams,
+      final RelationshipQueryParams queryParams,
       @Nonnull PageParams pageParams) {
-    TypedQuery<Relationship> relationshipTypedQuery =
-        getRelationshipTypedQuery(trackedEntity, queryParams, pageParams);
 
     return getPage(
-        pageParams, getList(relationshipTypedQuery), () -> countRelationships(queryParams));
+        pageParams,
+        relationshipsList(trackedEntity, queryParams, pageParams),
+        () -> countRelationships(trackedEntity, queryParams));
   }
 
   @Override
   public Page<Relationship> getByEnrollment(
       Enrollment enrollment, RelationshipQueryParams queryParams, @Nonnull PageParams pageParams) {
-    TypedQuery<Relationship> relationshipTypedQuery =
-        getRelationshipTypedQuery(enrollment, queryParams, pageParams);
-
     return getPage(
-        pageParams, getList(relationshipTypedQuery), () -> countRelationships(queryParams));
+        pageParams,
+        relationshipsList(enrollment, queryParams, pageParams),
+        () -> countRelationships(enrollment, queryParams));
   }
 
   @Override
   public Page<Relationship> getByEvent(
       Event event, RelationshipQueryParams queryParams, @Nonnull PageParams pageParams) {
-    TypedQuery<Relationship> relationshipTypedQuery =
-        getRelationshipTypedQuery(event, queryParams, pageParams);
-
     return getPage(
-        pageParams, getList(relationshipTypedQuery), () -> countRelationships(queryParams));
+        pageParams,
+        relationshipsList(event, queryParams, pageParams),
+        () -> countRelationships(event, queryParams));
   }
 
-  private int countRelationships(RelationshipQueryParams queryParams) {
-    if (queryParams.getEntity() instanceof TrackedEntity te) {
-      return getByTrackedEntity(te, null).size();
-    }
+  /**
+   * Query to extract a relationships with the order by clause and pagination if required
+   *
+   * @param entity to filter the relationships
```

Comment:
```suggestion
   * @param entity to filter the relationships by
```

I think it's best to either fill out all the javadoc params or not add them at all as they just take up space with no added value.